src/easylink/step.py

```diff
@@ -150,6 +144,32 @@ def validate_step(self, step_config: LayeredConfigTree) -> Dict[str, List[str]]:
             ]
         return errors
 
+    def get_implementation_node_name(self, step_config):
+        """Resolve a sensible unique node name for the implementation graph.
+        This method compares the step node names with the step names through the step hierarchy and
+        uses the full suffix of step names starting from wherever the two first differ. For example,
+        if we have node names step_3_loop_1 step_3_python_pandas the resulting implementation node name
+        will be step_3_loop_1_step_3_python_pandas. If all the node names and step names match, we have not
+        introduced any step degeneracies with e.g. loops or multiples, and we can simply use the implementation
+        name."""
+        step = self
+        node_names = []
+        step_names = []
+        while step:
+            node_names.append(step.name)
+            step_names.append(step.step_name)
+            step = step.parent_step
+
+        implementation_names = []
+        step_names.reverse()
+        node_names.reverse()
+        for i, (step_name, node_name) in enumerate(zip(step_names, node_names)):
+            if step_name != node_name:
+                implementation_names = node_names[i:]
```

Comment:
I could use a talk-through of the logic here. It seems like this should set to `node_names[i+1:]`?

Reply:
I'm happy to talk at mid-day, but say you have implementations from two loops, call them 1 and 2. The step names and node names would be:
1)
step_names == ["step_3", "implementation_name"]
node_names == ["step_3_loop_1", "implementation_name"]
2)
step_names == ["step_3", "implementation_name"]
node_names == ["step_3_loop_2", "implementation_name"]
We want to avoid calling both implementations "implementation_name", which is what `node_names[i+1:]` would give. For that, we need to *include* the node name for which they differ from the step name, so that 1 is called "step_3_loop_1_implementation_name" and 2 is called "step_3_loop_2_implementation_name", which is what `node_names[i:]` gives.

Reply:
Are you sure you want break here and not continue?